packages/massa-web3/src/web3/accounts/Web3Account.ts

```diff
@@ -134,9 +141,18 @@ export class Web3Account extends BaseClient implements IBaseAccount {
       expiryPeriod,
     );
 
+    // Chain id is an 64-bit unsigned integer, convert to byte array (big endian)
+    const chainIdBuffer = new ArrayBuffer(8);
+    const view = new DataView(chainIdBuffer);
+    view.setBigUint64(0, this.chainId, false);
+
     // sign payload
     const signature: ISignature = await this.sign(
-      Buffer.concat([getBytesPublicKey(this.account.publicKey), bytesCompact]),
+      Buffer.concat([
+        Buffer.from(chainIdBuffer),
+        getBytesPublicKey(this.account.publicKey),
+        bytesCompact,
+      ]),
```

Comment:
we are duplicating a lot of code here, better write a function that prepare the buffer we want to sign.

Reply:
Fixed